File: 백준/Silver/3135. 라디오/라디오.py
current, destination = map(int, input().split())

n = int(input())

button = []

result = 0
flag = False
min = abs(current - destination)
for _ in range(n):
  btn = int(input())
  if btn == destination:
    result = 1
  button.append(btn)

if result < 1:
  for item in button:
    if min > abs(item - destination):
      min = abs(item - destination)
      flag = True
      
  if flag:
    print(min + 1)
  else:
    print(min)

else:
  print(result)

File: 백준/Silver/3135. 라디오/라디오.c
#include <stdio.h>
#include <stdlib.h>

int main(void) { 
   
   int destination, current, index;
   scanf("%d %d", &current, &destination);

   int n, flag = 0;
   scanf("%d", &n);

   int arr[n], count = 0, min = abs(current - destination);
   for(int i = 0; i < n; i++) {
      scanf("%d", &arr[i]);
      if(arr[i] == destination){
         count += 1;
      }
   }

   if(count == 0){
      for(int i = 0; i < n; i++){
         if(min > abs(arr[i] - destination)){
            min = abs(arr[i] - destination);
            index = i;
            flag = 1;
         }
      }
   }
   
   if(count > 0) printf("%d", count);
   else {
      if(flag)   printf("%d", min + 1);
      else   printf("%d", min);
   }
   return 0;
}